extends Node
class_name InvetorySystem

signal coin_changed(new_amount: int)
signal item_collected(item_type: String, amount: int)
signal key_changed(new_amount:int)
signal health_restored(amount: int)
var coins: int = 0
var keys: int = 0
var fruit: int = 0
func _ready() -> void:
	pass

func add_coin(amount: int) -> void:
	coins += amount
	coin_changed.emit(coins)
	item_collected.emit("coin", amount)
	print("Collected ", amount, " coins. Total: ", coins)

func add_fruit(amount: int) -> void:
	fruit += amount # Tăng số lượng fruit
	item_collected.emit("fruit", amount)
	print("Collected ", amount, " fruit. Total: ", fruit)
	
	var health_amount: int = 20
	health_restored.emit(health_amount)
	print("Restored ", health_amount, " health.")
	# ==========================
		
	
func add_key(_amount: int = 1) -> void:
	keys += _amount
	key_changed.emit(keys)
	item_collected.emit("keys", _amount)
	print("Collected ", _amount, " keys. Total: ", keys)
	
	
func use_key(_amount: int = 1) -> bool:
	keys -=_amount
	return false

func has_key() -> bool:
	return keys > 0	

func get_gold() -> int:
	return coins

func get_keys() -> int:
	return keys
